lib/validator.ts

```diff
@@ -0,0 +1,101 @@
+import * as Axios from "axios";
+import { verify } from "jsonwebtoken";
+
+// tslint:disable-next-line:no-var-requires
+const jwkToPem = require("jwk-to-pem");
+
+import {
+    IJwtValidatorConfig,
+    IMapOfKidToPublicKey,
+    IPublicKeys,
+    ITokenHeader,
+    IClaim,
+    IValidationOptions,
+    IValidateTokenResult,
+} from "../declarations/declarations";
+import { DefaultValidationOptions } from "./constants";
+
+export class JwtValidator {
+    private validationOptions: IValidationOptions;
+    private cacheKeys?: IMapOfKidToPublicKey;
+
+    constructor(
+        private config: IJwtValidatorConfig,
+        validationOptions?: IValidationOptions,
+    ) {
+        this.validationOptions = Object.assign({}, DefaultValidationOptions, validationOptions || {});
```

Comment:
Defaulting to `{}` is unnecessary

```suggestion
        this.validationOptions = Object.assign({}, DefaultValidationOptions, validationOptions);
```